File: 03_recursion/countdown.php
<?php

/*
 * Recursão - É quando uma função chama a si mesma
 * Utilizada para tornar a resposta mais clara.
 *
 * */

function countdown($i)
{
    echo $i;

    if ($i <= 0) {
        return;
    } else {
        countdown($i - 1);
    }
}

countdown(10);

File: 03-Recursion/countdown/Countdown.java
public class Countdown {

	private static void countdown(int number) {

		System.out.println(number);

		if (number <= 0) {
			return;
		} else {
			countdown(number - 1);
		}
	}

	public static void main(String[] args) {
		
		countdown(10);
	}
}